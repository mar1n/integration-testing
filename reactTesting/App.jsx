import React from "react";

export const App = () => {
    const [cheesecakes, setCheesecakes] = React.useState(0);

    return(
        <div>
            <h1>Inventory Contents</h1>
            <p>Cheesecakes: {cheesecakes}</p>
            <button onClick={() => setCheesecakes(cheesecakes + 1)}>
                Add Cheesecake
            </button>
        </div>
    )
}