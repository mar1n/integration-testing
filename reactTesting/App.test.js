import React from "react";
import { App } from "./App.jsx";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import { screen } from "@testing-library/dom";

const root = document.createElement("div");
document.body.appendChild(root);

test('renders the approprate header', () => {
    act(() => {
        render(<App />, root);
    });
    expect(screen.getByText("Inventory Contents")).toBeTruthy();
});