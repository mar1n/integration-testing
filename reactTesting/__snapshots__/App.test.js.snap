// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updating the action log when loading items 1`] = `
<div
  data-testid="action-log"
>
  <h2>
    Action Log
  </h2>
  <ul>
    <li>
      Date: 
      Sat, 20 Jun 2020 13:37:00 GMT
       -
       
      Message: 
      Loaded items from the server
       - 
       
      Data: 
      {"status":200,"body":{"cheesecake":2,"croissant":5,"macaroon":96}}
    </li>
  </ul>
</div>
`;
